src/k8s/pkg/k8sd/api/component.go

```diff
@@ -109,3 +114,28 @@ func storageComponentPut(s *state.State, r *http.Request) response.Response {
 	}
 	return response.SyncResponse(true, &api.UpdateDNSComponentResponse{})
 }
+
+func ingressComponentPut(s *state.State, r *http.Request) response.Response {
+	var req api.UpdateIngressComponentRequest
+	snap := snap.SnapFromContext(s.Context)
+
+	err := json.NewDecoder(r.Body).Decode(&req)
+	if err != nil {
+		return response.SmartError(fmt.Errorf("failed to decode request: %w", err))
+	}
+
+	if req.Status == api.ComponentEnable {
+		err = component.EnableIngressComponent(
+			snap,
+			req.Config.DefaultTLSSecret,
+			req.Config.EnableProxyProtocol,
+		)
+	} else {
+		err = component.DisableIngressComponent(snap)
+	}
+	if err != nil {
+		return response.SmartError(fmt.Errorf("failed to %s %s: %w", req.Status, "ingress", err))
+	}
```

Comment:
maybe make this a switch-case for `api.ComponentEnable`, `api.ComponentDisable` and a `default:` clause. let's be clear if the user specifies a wrong value here.

also, `req.Status` will be `enabled/disabled` which does not read very well on the error message. Perhaps we can bite the bullet and have two error checks on each case above.